feat(sonic): restructuring page (#11132)
newrelic/docs-website: src/content/docs/logs/forward-logs/forward-your-logs-using-infrastructure-agent.mdx

import osAnsibleRed from 'images/os_icon_ansible-red.png'

> Forwarding your logs to New Relic makes all of your logging data available in one location, providing deeper visibility into both your application and your platform performance data. With your logs in one spot, you get enhanced log management capabilities that let you collect, process, explore, query, and alert on errors or anomalies found in your log data. 

Author: ```suggestion
Forwarding your logs to New Relic makes all of your logging data available in one location, providing deeper visibility into both your application and your platform performance data. With your logs in one spot, you can collect, process, explore, query, and alert on errors or anomalies found in your log data. 
```

> Your first step is to [create a New Relic account](). After that, use this doc to set up log forwarding. 

Reviewer: Not sure if the link is supposed to be missing here, just doing due diligence so we have a marker.

Author: LOL omg thank you.

/>

<figcaption>
From your host's UI, your logs are placed in context of events for the selected time period. You can drill down into detailed data for any of the highlighted attributes.
</figcaption>

How you forward your logs depends on how you installed the infrastructure agent, since the infrastructure agent enables the log forwarding feature. You can install the infrastructure agent via:

* Guided install (recommended for most users)
* Manual installation
* Linux tarball

<CollapserGroup>
<Collapser
    className="freq-link"
    id="system"
    title="Review system requirements"
  >

* Infrastructure agent version 1.11.4 or higher
* [Fluent Bit](https://fluentbit.io/). The infrastructure agent already installs the latest version for you. To update or downgrade it to a specfic version, refer to the [Fluent Bit installation](#install-fb-version) procedures.
* OpenSSL library 1.1.0 or higher
* Built-in support for ARM64 architecture on Linux systems (for example, AWS Graviton architecture) added in infrastructure agent [1.20.6](https://github.com/newrelic/infrastructure-agent/releases/tag/1.20.6).
* Amazon Linux 2
* CentOS version 7 or higher

> * Debian version 9 ("Stretch") or higher. Version 11 is not supported.

Reviewer: This isn't a big deal at all I don't think but the period makes the punctuation inconsistent. Just doing due diligence.

/>

<figcaption>
From your host's UI, your logs are placed in context of events for the selected time period. You can drill down into detailed data for any of the highlighted attributes.
</figcaption>

## Install the infrastructure agent [#infra]

> Starting with version 1.11.4, the infrastructure agent forwards logs to New Relic. You can install the infrastructure agent with our guided install, or with a manual install. Our guided install is a lightweight route that installs our infrastructure agent, an APM agent, and more. This is a good choice for smaller organizations or any organization that wants to test out New Relic. 

Reviewer: Do we need to put the version where this feature started here? Just asking for curiosity's sake

Reviewer: I don't think you need a comma in the second sentence. Something like "guided or manual install" might work better and eliminates minor repetition.

  />
</TechTileGrid>

If you don't have a New Relic account yet, or if you prefer to follow the procedure manually, see our [tutorial to install the package manager](/docs/infrastructure/install-infrastructure-agent/linux-installation/install-infrastructure-agent-linux-using-package-manager), or check out our [MSI installer](/docs/infrastructure/install-configure-manage-infrastructure/windows-installation/install-infrastructure-windows-server-using-msi-installer) (Windows).

<Callout variant="important">
  The log forwarding feature is not included when the infrastructure agent is implemented using Linux tarball or Windows ZIP installations.
</Callout>

## Configure the infrastructure agent [#configure]

> Configuration files direct your system to forward the log sources you want to appear in New Relic. You can add as many config files as you want. Our infrastructure agent uses `.yml` files to configure logging. If you install the infrastructure agent via [Add data](https://one.newrelic.com/marketplace?state=78678a7f-91c5-ca40-ac55-e6b74a50085c) in the UI, the file `logging.yml` is created automatically.

Reviewer: I like this change!

2. Create a `logging.yml` configuration file, and add the parameters you need. The `logging.d` directory has various `.yml.example` files you can use as a reference or starting point.

The agent automatically processes new configuration files without having to restart the infrastructure monitoring service. The only exception to this is when configuring a custom Fluent Bit configuration.

> ## Log forwarding parameters [#parameters]

Author: Maybe more action oriented title

> You're required to set up a `name` and log source parameter in the log forwarding `.yml` config file. To start, define a `name` of the log or logs you want to forward to New Relic. 

Reviewer: Feels a bit disjointed to have name in code but log source parameter not. Not sure if the fix is both being in code format or neither.

### Optional configuration [#optional-config]

> Although these configuration parameter's aren't required, we still recommend you set them up so you get the most functionality out of log forwarding.

Author: ```suggestion
Although these configuration parameters aren't required, we still recommend you set them up so you get the most functionality out of log forwarding.
```

    **Example:**

    ```
    sudo cp /etc/newrelic-infra/logging.d/nginx-log.yml.example /etc/newrelic-infra/logging.d/nginx-log.yml
    ```
  </Collapser>

  <Collapser
    id="rabbitmq-logs"
    title="Rabbitmq logs"
  >
    Copy (or rename) the `rabbitmq-log.yml.example` file to `rabbitmq-log.yml` to enable automatic Rabbitmq error log parsing and forwarding to New Relic. No need to restart the agent.

    **Example:**

    ```
    sudo cp /etc/newrelic-infra/logging.d/rabbitmq-log.yml.example /etc/newrelic-infra/logging.d/rabbitmq-log.yml
    ```
  </Collapser>

  <Collapser
    id="redis-logs"
    title="Redis logs"
  >
    Copy (or rename) the `redis-log.yml.example` file to `redis-log.yml` to enable automatic Redis error log parsing and forwarding to New Relic. No need to restart the agent.

    **Example:**

    ```
    sudo cp /etc/newrelic-infra/logging.d/redis-log.yml.example /etc/newrelic-infra/logging.d/redis-log.yml
    ```
  </Collapser>
</CollapserGroup>

## Enable log forwarding on agent installed using Linux tarball [#tarball-install]

> Our custom Linux installation process for infrastructure monitoring allows you to tailor all aspects of the installation process, and to place files and folders on your machine. If you choose the [assisted](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/linux-installation/tarball-assisted-install-infrastructure-agent-linux) or [manual](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/linux-installation/tarball-manual-install-infrastructure-agent-linux) tarball installation process, follow these steps to implement the log forwarder feature:

Reviewer: I think the links are formatted incorrectly but I'm not sure if they need to be full or start at /docs/ for some specific reason.

1. Create the following directories:

- `/var/db/newrelic-infra/newrelic-integrations/logging`
- `/etc/newrelic-infra/logging.d`

2. Download and install New Relic's [fluent-bit-package (RPM)](https://github.com/newrelic/fluent-bit-package/releases) by running a command similar to:
  ```shell
   yum localinstall td-agent-bit-<some-version>.rpm`
   ```

3. Download New Relic's [fluentbit plugin](https://github.com/newrelic/newrelic-fluent-bit-output/releases) and save it as `/var/db/newrelic-infra/newrelic-integrations/logging/out_newrelic.so`.

> 4. Dowload or copy the `parsers.conf` file from [this Github repository](https://github.com/newrelic/fluent-bit-package/blob/main/parsers.conf), and save it as `/var/db/newrelic-infra/newrelic-integrations/logging/parsers.conf`.

Reviewer: Download misspelled. Should Vale have caught this?
